Automattic/wp-calypso
# Add tracks to the Blaze Dashboard app

We use some tracks in the Advertising dashboard to find errors in the Campaign creation flow. I am now enabling these tracks in the Blaze Dashboard app.

## Proposed Changes

* Use the exiting `analyticsMiddleware` inside the Blaze app

## Testing Instructions

You will need a jetpack local environment running to test this. Also, you will need to be able to execute a code snippet that will open a gate for the Blaze Dashboard. To do so, follow these instructions:
* Install the Plugin Code Snippets
* Create a new snippet with the following content: `add_filter( 'jetpack_blaze_dashboard_enable', '__return_true' );`

---

* Checkout this branch, then open a terminal and navigate to wp-calypso/apps/blaze-dashboard
* Execute BLAZE_DASHBOARD_PACKAGE_PATH=/path/to/jetpack/projects/packages/blaze yarn dev
You will need to change the path to your local Jetpack installation.
* In another terminal tab, go to the Jetpack repo directory and execute: pnpm jetpack build packages/blaze
* Now open your Jurassic tunnel and enter the wp-admin page for your local site
* Click on Tools->Advertising
* In your browser, open your network tab, and show all request filtering by the name `pixel.wp.com/t.gif`
* Then, click on the Promote button in the header of the page
* Verify that the a correct track event is logged
![Screenshot 2023-06-29 at 14 05 34](https://github.com/Automattic/wp-calypso/assets/1258162/49780154-7fd4-4974-89f4-5c8978849fc1)

* Close the Create campaign widget, and then click on the Promote button of a post
* Verify that the correct track event is logged with the entry point `promoted_posts-post_item`

## Pre-merge Checklist
